### ISSUE TITLE
Milliseconds sometimes equals to 1000 and SQL Server doesn't like it

### ISSUE THREAD

**Issue by Jonathan57500** [briannesbitt/Carbon#1534]
Hello,

I encountered an issue with the following code:
```php
$str="";
        for($i=0;$i<20000;$i++)
        {
            $str .= Carbon::now()->format('Y-m-d H:i:s.v') . "<br/>";
        }
        return $str;
```
Carbon version: 1.36.1
PHP version: 7.1.24

I expected to get:
```
2018-11-23 17:48:34.999
2018-11-23 17:48:34.999
2018-11-23 17:48:35.000
2018-11-23 17:48:35.000
```
But I actually get:
```
2018-11-23 17:48:34.999
2018-11-23 17:48:34.999
2018-11-23 17:48:34.1000
2018-11-23 17:48:34.1000
...
2018-11-23 17:48:34.1000
2018-11-23 17:48:34.1000
2018-11-23 17:48:35.000
2018-11-23 17:48:35.000
```
The problem is that if I insert a record into a database MS SQL Server with Carbon::Now() on a Datetime column when millisecond = 1000 the insert fails.

Thanks!

**Comment by kylekatarnls** [comment]
Hi, this is a known PHP bug: `(new DateTime())->format('v')` sometimes return `'1000'`, when you call `format`, you directly call the `DateTime::format` method. We should not intercept format result, it would be very risky and slow.

It's a bad rounding of the microseconds so I recommend to use `u` format (microseconds) then truncate with something like:

```php
preg_replace('/(\.\d{3})\d+$/', '$1', Carbon::now()->format('Y-m-d H:i:s.u'));
```

**Comment by Jonathan57500** [comment]
Hi,
OK I understand. If it is a known issue of PHP, we are not able to know when it will be fixed. Is it possible for Carbon to add a test on Carbon::now() method (and all others that could return a millisecond value = 1000) or something else to temporary fix this bug ?

**Comment by Jonathan57500** [comment]
It is not possible to upgrade to Carbon 2 as Carbon is now part of Laravel Framework:
https://github.com/laravel/framework/issues/24961
They need to improve compatibility before Laravel's requirements upgrade Carbon from 1.26.x to 2.6.x...
I'll use your preg_replace solution, not the best but easier right now. Or may be upgrade to PHP 7.3 as this issue seems to be fixed.
Thanks again

**Comment by kylekatarnls** [comment]
Wrong. Laravel 5.8 yet use Carbon 2. Other version support it even if they require Carbon 1 as Carbon 2 did not exist when they have been released.

It's possible to upgrade despite your other dependencies restrictions and it's explained in the doc since the Carbon 2 first stable release: https://carbon.nesbot.com/

And it's also explained in the issue you mention.
